invenio_checks/components.py

```diff
@@ -21,31 +24,58 @@ def enabled(self):
         """Return if checks are enabled."""
         return current_app.config.get("CHECKS_ENABLED", False)
 
-    def update_draft(self, identity, data=None, record=None, errors=None, **kwargs):
+    def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
         """Update handler."""
         errors = errors or []
 
         if not self.enabled:
             return
 
         communities = []
-        if record.review.is_open:
-            communities.append(record.review.community)
+        if record.parent.review and (
+            record.parent.review.status == "submitted"
+            or record.parent.review.status == "created"
+        ):
+            # drafts can only be submitted to one community
+            communities.append(record.parent.review.receiver.resolve().id)
+        else:
+            results = current_requests_service.search(
+                identity,
+                extra_filter=dsl.query.Bool(
+                    "must",
+                    must=[
+                        dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                        dsl.Q("term", **{"is_open": True}),
+                    ],
+                ),
+            )
 
-        communities.extend(record.parent.communities.entries)
-        checks = itertools.chain(*[c.checks for c in communities])
+            if results.total > 0:
+                for result in results:
+                    communities.append(result.get("receiver", {}).get("community"))
```

Comment:
Given the comment from @slint above, I don't know if you have decided to go with a search in the end.
In any case, in general, without paginating or using `scan` you are only going to get the first `n` results.

Moreover, I would recommend avoiding using optionals as much as possible. This:
```
                    communities.append(result.get("receiver", {}).get("community"))
```
might add a `None` value to the list.
You might want to use a `set()` to avoid duplicates, and use `result["receiver"]["community"]`, or check its value before appending.

Reply:
Opened https://github.com/inveniosoftware/invenio-checks/issues/21 for addressing the requests search issue. For the uniqueness and optional checking this has been fixed here.

Reply:
Good, however you did not solve the `.get` potential issue... just in case you missed it ;)